clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -2565,6 +2574,12 @@ void cir::FuncOp::print(OpAsmPrinter &p) {
 
   auto cirVisibilityAttr = getGlobalVisibilityAttr();
   printVisibilityAttr(p, cirVisibilityAttr);
+  // TODO: This is a problematic space to be handled conditionally by
+  // printVisibilityAttr which leads often to a double space in the output. But
+  // it looks like from here we have also switched from adding a conditional
+  // trailing space to inserting a leading space, to avoid trailing space at
+  // EOL.
```

Comment:
The extra whitespace is an unfortunate bug introduced when visibility attribute got added, we should fix that at some point too.